src/main/java/duke/task/TaskList.java

```diff
@@ -0,0 +1,101 @@
+package duke.task;
+
+import java.util.ArrayList;
+import duke.exception.DukeTaskDetailsException;
+import duke.exception.DukeIndexInputException;
+
+/**
+ * Class that represents the list of tasks
+ */
+public class TaskList {
+
+    private ArrayList<Task> tasks;
+
+    public TaskList(ArrayList<Task> tasks) {
+        this.tasks = tasks;
+    }
+	
+    /**
+     * Method for adding task to list given user input, returns the added task if successful
+     * throws DukeTaskDetailsException otherwise
+     *
+     * @param taskArray String[] of user input split once by " "
+     *
+     * @return Task that was added if successful
+     */
+    public Task addTask(String[] taskArray) throws DukeTaskDetailsException {
+        Task task;
+        if (taskArray.length < 2) {
+            throw new DukeTaskDetailsException("Please provide task details");
+        }
+        if (taskArray[0].equals("todo")) {
+            task = new Todo(taskArray[1]);
+        } else if (taskArray[0].equals("deadline")) {
+            String[] deadlineDetails = taskArray[1].split("/by");
+            if (deadlineDetails.length != 2) {
+                throw new DukeTaskDetailsException("Please provide both task title and deadline, separated by \"/by\"");
+            }
+            task = new Deadline(deadlineDetails[0], deadlineDetails[1]);
+        } else {
+            String[] eventDetails = taskArray[1].split("/at");
+            if (eventDetails.length != 2) {
+                throw new DukeTaskDetailsException("Please provide both event title and date, separated by \"/at\"");
+            }
+            task = new Event(eventDetails[0], eventDetails[1]);
+        }
+        this.tasks.add(task);
+        return task;
+    }
+
+    /**
+     * Method for index commands such as "done" and "delete". Returns task if command
+     * was succesful, or throw DukeIndexInputException otherwise
+     *
+     * @param taskArray String[] of user input split once by " "
+     * 
+     * @return Task that was operated on if successful
+     */
+    public Task indexCommand(String[] taskArray) throws DukeIndexInputException {
+        if (taskArray.length < 2) {
+            throw new DukeIndexInputException("Please enter index of the task");
+        }
+        try {
+            Task task;
+            int index = Integer.parseInt(taskArray[1]);
+            if (taskArray[0].equals("done")) {
+                this.tasks.get(index - 1).markAsDone();
+                task = this.tasks.get(index - 1); 
+            } else {
+                task = this.tasks.remove(index - 1);
+            }
+            return task;
+        } catch (NumberFormatException e){
+            throw new DukeIndexInputException("Please enter index of the task");
```

Comment:
I like your error messages! Clearly explaining what the user should be doing instead!